crates/rpc/src/lib.rs

```diff
@@ -380,15 +380,43 @@ impl Debug for PeerAddr {
     }
 }
 
-#[derive(Clone, Copy, Debug, TryFrom)]
-#[try_from(repr)]
-#[repr(i32)]
-enum ConnectionStatusCode {
-    Ok = 0,
+#[derive(Clone, Copy, Debug)]
+enum ConnectionStatus {
+    Ok,
 
-    UnsupportedProtocol = -1,
-    UnknownApi = -2,
-    Unauthorized = -3,
+    UnsupportedProtocol,
+    UnknownApi,
+    Unauthorized,
+
+    Rejected { reason: u8 },
+}
+
+impl ConnectionStatus {
+    fn try_from(code: i32) -> Option<Self> {
+        Some(match code {
+            0 => Self::Ok,
```

Comment:
What are positive numbers reserved for? Why not simplify these codes to use positive numbers as a rejection reason, while negative numbers would be used for other variants of the enum.

Reply:
As this is a "status" code, and not an "error" code, error variants should be negative.
Potentially there may be other successful status codes.

Now it's not that useful, so it could have been `ConnectionError` with variants starting from `1`, and we would map it to `i32`, but that that didn't look much better.